### PR TITLE
Create TeamsFilesUploaded.yaml

### PR DESCRIPTION
**Pull request by thmcelro**
Create hunting query to identify files sent over teams and provide a summary of users that have accessed them.

### CODE REVIEW

File: Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - T1078
+query: |
+
+  OfficeActivity 
+  | where RecordType =~ "SharePointFileOperation" 
+  | where SourceRelativeUrl contains "Microsoft Teams Chat Files" 
+  | where Operation =~ "FileUploaded" 
+  | join kind= leftouter ( 
+     OfficeActivity 
+      | where RecordType =~ "SharePointFileOperation"
```
Comment (petebryan) [inline Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml:25]: Consider excluding access attempts from "app@sharepoint" which is a SharePoint farms service account and will generate a lot of noise.

File: Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  | where SourceRelativeUrl contains "Microsoft Teams Chat Files" 
+  | where Operation =~ "FileUploaded" 
+  | join kind= leftouter ( 
+     OfficeActivity 
+      | where RecordType =~ "SharePointFileOperation" 
+      | where SourceRelativeUrl contains "Microsoft Teams Chat Files" 
+      | where Operation =~ "FileDownloaded" or Operation =~ "FileAccessed" 
+  ) on OfficeObjectId 
+  | extend userBag = pack(UserId1, ClientIP1) 
+  | summarize makeset(UserId1), make_bag(userBag) by TimeGenerated, UserId, OfficeObjectId, SourceFileName
```
Comment (petebryan) [inline Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml:30]: You could also add:
| extend NumberUsers = array_length(bag_keys(bag_userBag))
This gives a count of the number of users who have accessed the file, you can then project that and allow users to filter results by this number.

File: Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml (hunk 1 above)
Comment (thmcelro) [reply Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml]: This is now excluded

File: Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml (hunk 2 above)
Comment (thmcelro) [reply Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml:32]: Great addition, I have added this in!

File: Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  | where Operation =~ "FileUploaded" 
+  | join kind= leftouter ( 
+     OfficeActivity 
+      | where RecordType =~ "SharePointFileOperation" 
+      | where SourceRelativeUrl contains "Microsoft Teams Chat Files" 
+      | where Operation =~ "FileDownloaded" or Operation =~ "FileAccessed" 
+  ) on OfficeObjectId 
+  | extend userBag = pack(UserId1, ClientIP1) 
+  | summarize makeset(UserId1), make_bag(userBag) by TimeGenerated, UserId, OfficeObjectId, SourceFileName 
+  | project UploadTime=TimeGenerated, Uploader=UserId, FileLocation=OfficeObjectId, FileName=SourceFileName, AccessedBy=bag_userBag
```
[the remark this answers is not in the document]
Comment (thmcelro) [reply Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml]: I now use the custom entity and timestamp by default

**Pull request merged**